autoload/kite/client.vim

```diff
@@ -56,6 +57,17 @@ function! kite#client#onboarding_file(handler)
 endfunction
 
 
+function! kite#client#request_related(json, handler)
+  let path = s:codenav_path
+  if has('channel')
+    let response = s:internal_http(path, g:kite_short_timeout, a:json)
+  else
+    let response = s:external_http(s:base_url.path, g:kite_short_timeout, a:json)
+  endif
+  return a:handler(s:parse_response(response))
+endfunction
+
+
```

Comment:
This is synchronous, not asynchronous, which presumably is what you want?

Reply:
Oops, yep. Thanks for the pointers!